torchvision/utils.py

```diff
@@ -121,10 +126,74 @@ def save_image(
             If a file object was used instead of a filename, this parameter should always be used.
         **kwargs: Other arguments are documented in ``make_grid``.
     """
-    from PIL import Image
     grid = make_grid(tensor, nrow=nrow, padding=padding, pad_value=pad_value,
                      normalize=normalize, range=range, scale_each=scale_each)
     # Add 0.5 after unnormalizing to [0, 255] to round to nearest integer
     ndarr = grid.mul(255).add_(0.5).clamp_(0, 255).permute(1, 2, 0).to('cpu', torch.uint8).numpy()
     im = Image.fromarray(ndarr)
     im.save(fp, format=format)
+
+
+def draw_bounding_boxes(
+    image: torch.Tensor,
+    boxes: torch.Tensor,
+    labels: torch.Tensor,
+    label_names: List[str] = None,
+    colors: Dict[int, str] = None,
+    draw_labels: bool = True,
+    width: int = 1
+) -> torch.Tensor:
+
+    """
+    Draws bounding boxes on given image.
+
+    Args:
+        image (Tensor): Tensor of shape (C x H x W) or (1 x C x H x W)
+        bboxes (Tensor): Tensor of size (N, 4) containing bounding boxes in (xmin, ymin, xmax, ymax) format.
+        labels (Tensor): Tensor of size (N) Labels for each bounding boxes.
+        label_names (List): List containing labels excluding background.
+        colors (dict): Dict with key as label id and value as color name.
+        draw_labels (bool): If True (default) draws label names on bounding boxes.
+        width (int): Width of bounding box.
+    """
+
+    # Code co-contributed by sumanthratna
+
+    if not (torch.is_tensor(image)):
+        raise TypeError(f'tensor expected, got {type(image)}')
+
+    if(image.dim() == 4):
+        if(image.shape[0] == 1):
+            image = image.squeeze(0)
+        else:
+            raise ValueError("Batch size > 1 is not supported. Pass images with batch size 1 only")
+
+    if label_names is not None:
+        # Since for our detection models class 0 is background
+        label_names.insert(0, "__background__")
+
+    # Add 0.5 after unnormalizing to [0, 255] to round to nearest integer
+    ndarr = image.mul(255).add_(0.5).clamp_(0, 255).permute(1, 2, 0).to('cpu', torch.uint8).numpy()
+
+    # Neceassary check to remove grad if present
+    if(boxes.requires_grad):
+        boxes = boxes.detach()
+
+    boxes = boxes.to('cpu').numpy().astype('int').tolist()
+    labels = labels.to('cpu').numpy().astype('int').tolist()
+
+    img_to_draw = Image.fromarray(ndarr)
+    draw = ImageDraw.Draw(img_to_draw)
+
+    for bbox, label in zip(boxes, labels):
+        if colors is None:
+            draw.rectangle(bbox, width=width)
+        else:
+            draw.rectangle(bbox, width=width, outline=colors[label])
+
+        if label_names is None:
+            draw.text((bbox[0], bbox[1]), str(label))
+        elif draw_labels is True:
+            draw.text((bbox[0], bbox[1]), label_names[int(label)])
```

Comment:
Same comment here about having a single `draw.text` invocation, with a `**kwargs` that is passed through.